boot/bootutil/src/loader.c

```diff
@@ -1039,6 +1040,12 @@ boot_go(struct boot_rsp *rsp)
 
     switch (swap_type) {
     case BOOT_SWAP_TYPE_NONE:
+#ifdef BOOTUTIL_VALIDATE_SLOT0
```

Comment:
I guess this is more a nitpick than anything else, but what do you think about using `#if` here rather than `#ifdef`?  Undefined symbols resolve to 0 in preprocessor conditionals, so it works if the macro is not defined.  The plus side is that someone can define the symbol to 0 and the feature won't get enabled.